module_3/Dao/src/main/java/com/epam/ems/dto/Certificate.java

```diff
@@ -0,0 +1,107 @@
+package com.epam.ems.dto;
+
+import org.springframework.hateoas.RepresentationModel;
+
+import javax.persistence.*;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.Pattern;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+@Entity
+@Table(name = "epam.gift_certificate")
+public class Certificate extends RepresentationModel<Certificate> {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "certificate_id")
+    @Min(1)
+    private int id;
+
+    @Pattern(regexp = "^.{0,45}$")
+    private String name;
+    @Pattern(regexp = "^.{0,45}$")
+    private String description;
+    @Min(0)
+    private int price;
+    @Min(0)
+    private int duration;
+    @Column(name = "create_date")
+    @Pattern(regexp = "^.{0,45}$")
+    private String createDate;
+    @Column(name = "last_update_date")
+    @Pattern(regexp = "^.{0,45}$")
+    private String lastUpdateDate;
+
+    @ManyToMany(cascade = { CascadeType.ALL })
+    @JoinTable(
+            name = "epam.certificate_tag",
+            joinColumns = { @JoinColumn(name = "certificate_id") },
+            inverseJoinColumns = { @JoinColumn(name = "tag_id") }
+    )
+    private List<Tag> tags;
+
+    public Certificate() {
```

Comment:
Please use Lombok here
Annotate class with @Data and other annotation where needed to remove boilerplate code.